### ISSUE TITLE
Hide fallback email address

### ISSUE THREAD

**Issue by adam-fu** [simple-swiss-code/simple_mail_frontend#5]
Currently we list the fallback email as text. Have a look on ways to obfsucate:
[on stackoverflow](https://stackoverflow.com/questions/483212/effective-method-to-hide-email-from-spam-bots)
a more general overview can be found  on [moz.com](https://moz.com/blog/5-tips-for-hiding-your-email-address-from-spammers),  here is also a [overview](https://www.araweb.co.uk/Safe_email_how_to_protect_843) 
In general: ROT13 and changing "adam  [AT] example.com" is not a good technique. Spam bot harvesters are smarter than. Also Image will not work for our purpose. I ike the advanced scripting proposal in the last link and some techniques listed on [stackoverflow](https://www.araweb.co.uk/Safe_email_how_to_protect_843)

**Comment by nthungvn** [comment]
Hoi @adam-fu 

I read 2 general ideas link, and yep, for our case, we should use advanced scripting (i.e use JS combine with HTML and CSS if needed).

Some advanced solution in StackOverflow has its own pros and cons, so I listed some of them:

1. Use Javascript to concatenate all part of email like (no matter in dedicated JS file or embedded in HTML file)
`function getEmail(name, domain, topLevelDomain) {
}`
Refers to these answers: https://stackoverflow.com/a/10300743 or https://stackoverflow.com/a/41566570

With this solution, users can hover to the link and can see the email. Then if they want they can copy the email and manually open their preferred mail client.
But it's not strong and secure enough, we still expose parts of an email.

2. Put the encode email: embedded the defined character (e.g #) in the email as much as we can then we can use the JS to replace # by an empty string when using hover and click to the link.
`<a href="mailto:coxntact@domainx.com"
    onmouseover="this.href=this.href.replace(/x/g,'');">link</a>`

We can also encode the other characters with HTML code (https://stackoverflow.com/a/483225)

Refers to these answers: https://stackoverflow.com/a/26420534 and https://stackoverflow.com/a/483430 and https://stackoverflow.com/a/30476576

3. Combine with back-end solution, when clicking the link, the browser will request the backend and it will return and redirect to the default mail client

This solution, we completely protect our email and already applied in the http://tapetowaniewarszawa.pl/

Also refers to this answer: https://stackoverflow.com/a/3395273

Actually we can combine 1 and 2 together, or use only 3. Which one do you refer?

**Comment by adam-fu** [comment]
Hi Hung, yes I used solution 3 already, but in this case, we should show a fallback in case server does not accept the call. Maybe server is down?  So solution 3 is not applicable in the text

Our special case we show only the email if captcha was correct and server returned error. So we do not need to make any css tricks. We need just a bit reasonable protection from spam tools for reading and parsing  javascript source code for email addresses.
Of course as in the server solution we just need good enough. Any attacker will eventually get the email if they try hard enough. It should be enough to store the email as "encrypted" string in the code and run decode when loading content with jquery for the error page. 

I see we can use forge https://github.com/digitalbazaar/forge or other web crypt libraries for that and use the google token (if it sends us always the same one) as key. 
But is overill. So let us solve like that:

So basically its: 
1) use jQuery to write the email in the error response. Use atob and btoa is enough https://stackoverflow.com/questions/246801/how-can-you-encode-a-string-to-base64-in-javascript/247261#247261 
please provide a simple html outside of lib folder (root folder of this project is enough, for the users to store generate the email link for the online page,

**Comment by nthungvn** [comment]
Hoi Adam,

Yes. You were right.
Now I understand why we don't use the server-side solution. 👍 

Yep. So let using a simple solution.
For example, we need to prepare a encode email, for example, `encode.email@example.com` will have `ZW5jb2RlLmVtYWlsQGV4YW1wbGUuY29t`.

```
const encodeEmailSomeWhere = 'ZW5jb2RlLmVtYWlsQGV4YW1wbGUuY29t';
const decodeEmail = window.atob(encodeEmailSomeWhere);
// continue to use decodeEmail variable to build the clickable email sending in an anchor tag
```

Do you mean we need to read encodeEmail from file and this file will be HTML?
> please provide a simple HTML outside of lib folder (root folder of this project is enough, for the users to store generate the email link for the online page,

Or... something else? Because I thought, when we apply with `wallpaper_2020` project, we can use Liquid to fulfill the encodeEmail to our JS file from Jekyll configuration.

**Comment by adam-fu** [comment]
Yes, it goes into this direction.
But because we just show the email when captcha was solved successfuly, then
have btoa() outside the doc/ folder to encode the emial and then inside mail use atob() to get it human readable. atob and btoa() are default browser javascript methods. I think for yekyll its good to put it into configuration and run atob() at jekyll page build time and btoa() when the captcha was correct and server error

**Comment by nthungvn** [comment]
Hoi Adam,
I am not sure that I understand what you desire.
I will clarify some points:

>have btoa() outside the doc/ folder to encode the emial 

For simple_mail_frontend repo, why don't we hardcode the encode email in the variable of JS?
Because of
1. Reading the file in the static web app is not a native behaviour when running in the browser, and Github Pages only now refers to docs/ folder
2. I am not sure the JS file (inside docs/ folder) can read the file outside docs/ folder.

>and run atob() at jekyll page build time

I am not sure we can run the btoa() function for encoding the pure email or whatever JS function in the build time of Jekyll.
Currently, I hard-coded the encode email in the _config.yml
![image](https://user-images.githubusercontent.com/9164553/83324022-95c41c00-a28c-11ea-9bbc-5e5211808e88.png)

**Comment by nthungvn** [comment]
>I am not sure the JS file (inside docs/ folder) can read the file outside docs/ folder.

Clarified already 😆  see the screenshot below.

![image](https://user-images.githubusercontent.com/9164553/83328487-81dce200-a2ad-11ea-8728-b129a5c4c977.png)
